### PR TITLE
GEOT-5179, single layer z order control

### PR DESCRIPTION
**Pull request by aaime**
Single layer portion of GEOT-5179, and https://github.com/geotools/geotools/wiki/Z-order-control-during-map-production

### CODE REVIEW

**Comment by aaime** [comment]
Btw, please review this one first, I won't make a pull request for cross layer z ordering until this one, much simpler and with little/no refactoring required, is merged.

File: modules/library/main/src/main/java/org/geotools/filter/SortByImpl.java [hunk 1]
```diff
@@ -52,4 +52,55 @@ public void setSortOrder(SortOrder sortOrder) {
...
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see java.lang.Object#toString()
+     */
+    @Override
+    public String toString() {
+        return "SortByImpl [propertyName=" + propertyName + ", sortOrder=" + sortOrder + "]";
+    }
```
Comment (jodygarnett) [inline modules/library/main/src/main/java/org/geotools/filter/SortByImpl.java]: Thanks, toString() methods help so much when debugging

File: modules/library/render/src/test/java/org/geotools/renderer/lite/ZOrderTest.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
+package org.geotools.renderer.lite;
```
Comment (jodygarnett) [inline modules/library/render/src/test/java/org/geotools/renderer/lite/ZOrderTest.java]: Will add/fix headers during merge

Comment (aaime) [inline modules/library/render/src/test/java/org/geotools/renderer/lite/ZOrderTest.java]: No problem, just tell me where the problems are, I'll add the header

**Comment by jodygarnett** [comment]
merged, with docs, to master

**Pull request closed without merging**